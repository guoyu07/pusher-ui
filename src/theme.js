const fontFamilies = [
  '-apple-system',
  'BlinkMacSystemFont',
  'Segoe UI',
  'Roboto',
  'Oxygen',
  'Ubuntu',
  'Cantarell',
  'Fira Sans',
  'Droid Sans',
  'Helvetica Neue',
  'sans-serif',
].map(name => `"${name}"`).join(', ');


export const fontFamily = fontFamilies;
export const white = '#fff';
export const black = '#1e1e1e';
export const grey = '#e7ebf3';
export const darkGrey = '#d6d6d6';
export const darkDarkGrey = '#989898';
export const lightGrey = '#e7ebf3';
export const primary = '#549eff';
export const success = '#319a31';
export const warning = '#ffaf1c';
export const danger = '#ca0808';
export const zIndex1 = 1;
export const zIndex2 = 2;
export default {
  fontFamily,
  white,
  black,
  grey,
  darkGrey,
  darkDarkGrey,
  lightGrey,
  primary,
  success,
  warning,
  danger,
  zIndex1,
  zIndex2,
};
