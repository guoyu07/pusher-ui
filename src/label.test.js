import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'glamorous';

import { Label } from './index';
import * as theme from './theme';

describe('Label', () => {
  const defaultProps = {
    children: 'Hello, label',
  };
  const createSnapshot = (props = defaultProps) => {
    expect(
      renderer
        .create(
          <ThemeProvider theme={theme}>
            <Label {...props} />
          </ThemeProvider>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  };

  it('should render a label', () => {
    createSnapshot();
  });

  it('should have a different color if an error prop is set', () => {
    createSnapshot({ error: true });
  });

  it('should render children', () => {
    createSnapshot({ children: <div>Inner content</div> });
  });
});
