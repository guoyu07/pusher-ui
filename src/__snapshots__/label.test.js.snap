// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should have a different color if an error prop is set 1`] = `
.glamor-0,
[data-glamor-0] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  font-size: 0.75rem;
  letter-spacing: 0.0625rem;
  text-transform: uppercase;
  color: #ff455c;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

.glamor-0 input,
[data-glamor-0] input,
.glamor-0 textarea,
[data-glamor-0] textarea {
  margin-top: 0.375rem;
}

<label
  className="glamor-0"
/>
`;

exports[`Label should render a label 1`] = `
.glamor-0,
[data-glamor-0] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  font-size: 0.75rem;
  letter-spacing: 0.0625rem;
  text-transform: uppercase;
  color: #95979d;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

.glamor-0 input,
[data-glamor-0] input,
.glamor-0 textarea,
[data-glamor-0] textarea {
  margin-top: 0.375rem;
}

<label
  className="glamor-0"
>
  Hello, label
</label>
`;

exports[`Label should render children 1`] = `
.glamor-0,
[data-glamor-0] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  font-size: 0.75rem;
  letter-spacing: 0.0625rem;
  text-transform: uppercase;
  color: #95979d;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

.glamor-0 input,
[data-glamor-0] input,
.glamor-0 textarea,
[data-glamor-0] textarea {
  margin-top: 0.375rem;
}

<label
  className="glamor-0"
>
  <div>
    Inner content
  </div>
</label>
`;
